packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -99,6 +98,147 @@ export function TelemetryView(): React.ReactElement {
 		};
 	}, [messageRelay, setTelemetryEvents]);
 
+	return (
+		<Stack>
+			<StackItem>
+				<ListLengthSelection
+					currentLimit={maxEventsToDisplay}
+					onChangeSelection={(key): void => setMaxEventsToDisplay(key)}
+				/>
+			</StackItem>
+			<StackItem>
+				{telemetryEvents !== undefined ? (
+					<FilteredTelemetryView telemetryEvents={telemetryEvents} />
+				) : (
+					<Waiting label={"Waiting for Telemetry events"} />
+				)}
+			</StackItem>
+		</Stack>
+	);
+}
+
+/**
+ * {@link ListLengthSelection} input props.
+ */
+interface ListLengthSelectionProps {
+	/**
+	 * The current limit (max number of telemetry events to show).
+	 * @defaultValue {@link DEFAULT_PAGE_SIZE}
```

Comment:
Nit: This property isn't optional, so I don't think the defaultValue comment here adds value. This is policy set by the consumer. Generally `@defaultValue` accompanies optional properties.